src/main/java/com/snowplowanalytics/snowplow/tracker/emitter/EventStore.java

```diff
@@ -9,12 +9,9 @@
 
     boolean add(TrackerEvent trackerEvent);
 
-    void removeAllEvents(List<TrackerEvent> eventsList);
+    void removeEvents(List<TrackerEvent> eventsList, int numberToRemove);
```

Comment:
I also think this is particularly dangerous in the case of the InMemoryEventStore. Using `drainTo` on a list you don't "control" is risky because if that list changes during the `drainTo` operation then an exception will be thrown. I think you'd be better to create a new list in the `removeEvents` method of the `InMemoryEventStore` and return that, as Alex's updated interface suggests.